pkg/controller/podgroup_test.go

```diff
@@ -183,12 +183,98 @@ func Test_Run(t *testing.T) {
 
 }
 
-func makePods(podNames []string, pgName string, phase v1.PodPhase) []*v1.Pod {
+func TestFillGroupStatusOccupied(t *testing.T) {
+	ctx := context.TODO()
+	cases := []struct {
+		name                 string
+		pgName               string
+		minMember            int32
+		podNames             []string
+		podPhase             v1.PodPhase
+		podOwnerReference    []metav1.OwnerReference
+		groupPhase           v1alpha1.PodGroupPhase
+		desiredGroupOccupied string
+	}{
+		{
+			name:      "fill the desired Occupied",
+			pgName:    "pg",
+			minMember: 2,
+			podNames:  []string{"pod1", "pod2"},
+			podPhase:  v1.PodPending,
+			podOwnerReference: []metav1.OwnerReference{
+				{
+					Name: "new-occupied",
+				},
+			},
+			groupPhase:           v1alpha1.PodGroupPending,
+			desiredGroupOccupied: "default/new-occupied",
+		},
+		{
+			name:      "fill the desired Occupied, with multi ownerReference",
+			pgName:    "pg",
+			minMember: 2,
+			podNames:  []string{"pod1", "pod2"},
+			podPhase:  v1.PodPending,
+			podOwnerReference: []metav1.OwnerReference{
+				{
+					Name: "new-occupied-1",
+				},
+				{
+					Name: "new-occupied-2",
+				},
+			},
+			groupPhase:           v1alpha1.PodGroupPending,
+			desiredGroupOccupied: "default/new-occupied-1,default/new-occupied-2",
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			var kubeClient *fake.Clientset
+			if len(c.podNames) == 0 {
+				kubeClient = fake.NewSimpleClientset()
+			} else {
+				ps := makePods(c.podNames, c.pgName, c.podPhase, c.podOwnerReference)
+				kubeClient = fake.NewSimpleClientset(ps[0], ps[1])
+			}
+			pg := makePG(c.pgName, c.minMember, c.groupPhase, nil)
+			pgClient := pgfake.NewSimpleClientset(pg)
+
+			informerFactory := informers.NewSharedInformerFactory(kubeClient, controller.NoResyncPeriodFunc())
+			pgInformerFactory := schedinformer.NewSharedInformerFactory(pgClient, controller.NoResyncPeriodFunc())
+			podInformer := informerFactory.Core().V1().Pods()
+			pgInformer := pgInformerFactory.Scheduling().V1alpha1().PodGroups()
+			ctrl := NewPodGroupController(kubeClient, pgInformer, podInformer, pgClient)
+
+			pgInformerFactory.Start(ctx.Done())
+			informerFactory.Start(ctx.Done())
+
+			go ctrl.Run(1, ctx.Done())
+			err := wait.Poll(200*time.Millisecond, 1*time.Second, func() (done bool, err error) {
+				pg, err := pgClient.SchedulingV1alpha1().PodGroups("default").Get(ctx, c.pgName, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				if pg.Status.OccupiedBy != c.desiredGroupOccupied {
```

Comment:
We can change `desiredGroupOccupied` to `[]string` and compare them by `cmp.Diff`

Reply:
the type of `pg.Status.OccupiedBy` is `string`, thus `cmp.Diff` works? as we change `desiredGroupOccupied ` to `[]string `.

Reply:
O.. I forget the OccupiedBy is string....